app/models/application.rb

```diff
@@ -22,6 +22,8 @@ class Application < ApplicationRecord
 
   validates :team, :application_data, presence: true
 
+  before_validation :ensure_flags_unique
+
```

Comment:
Since `flags` is an attribute on `Application`, wouldn't it be easier to add a validator instead of the before hook?  And back it up with a database constraint.  
If that is possible at all, I am not totally sure about how the flags are set. 

If that ^ is not possible, maybe add an error message to the :ensure_flags_unique method? Because this will fail silently, right?

Reply:
Thank you for the feedback!

Hm, I'm afraid I don't understand the question. What I'm doing here is cleaning up, not validating. Maybe the problem is in the name? Maybe `cleanup_flags` would be better?

What do you thinK?

Reply:
LOL As I totally misunderstood this as cleaning up, that may be a better name :-)
(I don't have access to the issue you are linking.) 
So, is the goal accomplished when the cleaning up happens _only_  when you update or create a validation?

Reply:
Sweet! Thanks for confirming that this was a bad naming issue :) 

This is meant to be run before the record is validated. That is, the flags will automatically be cleaned up before the application is validated (and by extension, saved). 

Hope that makes sense!